<launch>
  <arg name="imu_enabled" default="true" />
  <arg name="imu_rate" default="5.0" />
  <arg name="battery_enabled" default="true" />
  <arg name="battery_rate" default="0.5" />
  <arg name="nav_enabled" default="true" />
  <arg name="nav_rate" default="5.0" />

  <group ns="nmea_if">

    <node pkg="topic_tools" type="relay" name="navsat_sentence_relay"
          args="/navsat/nmea_sentence /navsat/rx" />

    <!-- GPS time offset calculator -->
    <param name="time_offset_default" value="-14401.595" />
    <node pkg="heron_nmea" type="gps_time_offset_publisher" name="time_offset">
      <remap from="rx" to="/navsat/rx" />
    </node>

    <!-- Relay sentences from GPS receiver -->
    <node pkg="topic_tools" type="relay" name="nmea_sentence_relay" args="/navsat/rx tx">
    </node>

    <!-- Relay GPS vel messages (for NVG and NVR messages) -->
    <node pkg="topic_tools" type="relay" name="nmea_vel_relay" args="/navsat/vel navsat/vel">
    </node>

    <!-- Publishes $CPNVG and $CPNVR messages -->
    <node pkg="topic_tools" type="throttle" name="odom_throttle"
          args="messages /robot_pose_ekf/odom $(arg nav_rate) odom"
          if="$(arg nav_enabled)" />
    <remap from="navsat/enu_datum" to="/navsat/enu_datum" />
    <node pkg="enu" type="services" name="enu_services" />
    <node pkg="heron_nmea" type="nav_publisher" name="nmea_nav" />

    <!-- Publishes $CPIMU and $CPRCM messages -->
    <node pkg="topic_tools" type="throttle" name="imu_throttle"
          args="messages /imu/data_compass $(arg imu_rate) imu/data"
          if="$(arg imu_enabled)" />
    <node pkg="heron_nmea" type="imu_publisher" name="nmea_imu" />

    <!-- Publishes $CPRBS messages -->
    <node pkg="topic_tools" type="throttle" name="battery_throttle"
          args="messages /sense $(arg battery_rate) sense"
          if="$(arg battery_enabled)" />
    <node pkg="heron_nmea" type="battery_publisher" name="nmea_battery" />

    <!-- Receives $PYDIR, $PYDEV, and $PYDEP messages -->
    <node pkg="heron_nmea" type="command_publisher" name="nmea_commands">
      <remap from="cmd_drive" to="/cmd_drive" />
      <remap from="cmd_helm" to="/cmd_helm" />
      <remap from="cmd_course" to="/cmd_course" />
    </node>

    <!-- NMEA HERON interface -->
    <node pkg="nmea_comms" type="socket_node" name="socket_node">
      <!-- <remap from="nmea_sentence_out" to="tx" /> -->
      <!-- remap from="nmea_sentence" to="rx" /> -->
    </node>
  </group>
</launch>
